typescript/web/src/pages/auth/signin.tsx

```diff
@@ -1,30 +1,51 @@
+import { Heading, VStack } from "@chakra-ui/react";
+import { useSession } from "next-auth/react";
 import { useRouter } from "next/router";
-import React, { useCallback } from "react";
-import {
-  SignInModal,
-  SignInModalProvider,
-} from "../../components/auth-manager/signin-modal";
-import { CookieBanner } from "../../components/cookie-banner";
-import { Layout } from "../../components/layout";
-import { NavLogo } from "../../components/logo/nav-logo";
+import React, { useEffect } from "react";
+import { StringParam, useQueryParam } from "use-query-params";
+import { SignIn, SignInProvider } from "../../components/auth";
 import { Meta } from "../../components/meta";
 
-const LocalDatasetsIndexPage = () => {
+const useRedirectIfAuthenticated = (redirectUrl?: string) => {
+  const { status } = useSession();
   const router = useRouter();
-  const exitSignIn = useCallback(async () => {
-    await router.replace({ pathname: `/`, query: router.query });
-  }, [router]);
+  useEffect(() => {
+    if (status === "authenticated") {
+      router.replace(redirectUrl || `/`);
+    }
+  });
+};
+
+const MustLogInMessage = () => (
+  <Heading as="h3">Please sign-in to continue</Heading>
+);
+
+type BodyProps = {
+  redirectUrl?: string;
+};
+
+const Body = ({ redirectUrl }: BodyProps) => {
+  return (
+    <SignInProvider>
+      <VStack mt={10} mx="auto" maxW="4xl" spacing={10}>
+        {redirectUrl && <MustLogInMessage />}
+        <SignIn />
+      </VStack>
+    </SignInProvider>
+  );
+};
+
+const SignInPage = () => {
+  const [redirect] = useQueryParam("redirect", StringParam);
+  const redirectUrl = redirect ?? undefined;
+  const { status } = useSession();
+  useRedirectIfAuthenticated(redirectUrl);
   return (
     <>
       <Meta title="LabelFlow | Sign in" />
-      <CookieBanner />
-      <Layout breadcrumbs={[<NavLogo key={0} />]}>
-        <SignInModalProvider onClose={exitSignIn}>
-          <SignInModal />
-        </SignInModalProvider>
-      </Layout>
+      {status !== "unauthenticated" ? null : <Body redirectUrl={redirectUrl} />}
```

Comment:
Why not `{status === "unauthenticated" && <Body redirectUrl={redirectUrl} />}` ?

Reply:
Because, IIRC, before some intermediate changes it was:

```tsx
{status !== "unauthenticated" ? <LayoutSpinner /> : <Body redirectUrl={redirectUrl} />}
```

And I've just replaced the spinner with `null` instead of simplifying the whole line.
Good catch! I'll fix it.